pkg/cmd/cli/cmd/newapp.go

```diff
@@ -271,15 +274,56 @@ func setLabels(labels map[string]string, result *newcmd.AppResult) error {
 	return nil
 }
 
+// isInvalidTriggerError returns true if the given error is
+// a validation error that contains 'trigger' in its error message.
+// This error is returned from an older server when
+// it doesn't recognize a newer trigger type.
+func isInvalidTriggerError(err error) bool {
+	if !kapierrors.IsInvalid(err) {
+		return false
+	}
+	statusErr, ok := err.(*kapierrors.StatusError)
+	if !ok {
+		return false
+	}
+	return strings.Contains(statusErr.Status().Message, "trigger")
+}
+
+// retryBuildConfig determines if the given error is caused by an invalid trigger
+// error on a BuildConfig. If that is the case, it will remove all triggers with a
+// type that is not in the whitelist for an older server.
+func retryBuildConfig(info *resource.Info, err error) runtime.Object {
+	triggerTypeWhiteList := map[buildapi.BuildTriggerType]struct{}{
+		buildapi.GitHubWebHookBuildTriggerType:  struct{}{},
+		buildapi.GenericWebHookBuildTriggerType: struct{}{},
+		buildapi.ImageChangeBuildTriggerType:    struct{}{},
```

Comment:
ignore as in drop? or ignore as in allow? allowing means future code needs to be much more tolerant of nil parameter structs that would otherwise be required (e.g. a trigger type Foo requires a FooParams peer field be set, but an old server let in type="Foo" while dropping the unknown FooParams field)